keras_nlp/models/bart/bart_preprocessor_test.py

```diff
@@ -0,0 +1,194 @@
+        self.assertAllEqual(
+            x_out["encoder_token_ids"], [[0, 3, 4, 5, 3, 6, 2, 1, 1, 1]] * 4
+        )
+        self.assertAllEqual(
+            x_out["encoder_padding_mask"], [[1, 1, 1, 1, 1, 1, 1, 0, 0, 0]] * 4
+        )
+        self.assertAllEqual(
+            x_out["decoder_token_ids"], [[0, 7, 8, 9, 10, 11, 2, 1]] * 4
+        )
+        self.assertAllEqual(
+            x_out["decoder_padding_mask"], [[1, 1, 1, 1, 1, 1, 1, 0]] * 4
+        )
+        self.assertAllEqual(y_out, y)
+        self.assertAllEqual(sw_out, sw)
+
+    def test_tokenize_labeled_dataset(self):
+        x = {
+            "encoder_text": [" airplane at airport"] * 4,
+            "decoder_text": [" kohli is the best"] * 4,
+        }
+        y = tf.constant([1] * 4)
+        sw = tf.constant([1.0] * 4)
+        ds = tf.data.Dataset.from_tensor_slices((x, y, sw))
+        ds = ds.map(self.preprocessor)
+        x_out, y_out, sw_out = ds.batch(4).take(1).get_single_element()
+        self.assertAllEqual(
+            x_out["encoder_token_ids"], [[0, 3, 4, 5, 3, 6, 2, 1, 1, 1]] * 4
+        )
+        self.assertAllEqual(
+            x_out["encoder_padding_mask"], [[1, 1, 1, 1, 1, 1, 1, 0, 0, 0]] * 4
+        )
+        self.assertAllEqual(
+            x_out["decoder_token_ids"], [[0, 7, 8, 9, 10, 11, 2, 1]] * 4
+        )
+        self.assertAllEqual(
+            x_out["decoder_padding_mask"], [[1, 1, 1, 1, 1, 1, 1, 0]] * 4
+        )
+        self.assertAllEqual(y_out, y)
+        self.assertAllEqual(sw_out, sw)
+
+    def test_error_multi_segment_input(self):
+        input_data = {
+            "encoder_text": (
+                tf.constant([" airplane at airport"] * 2),
+                tf.constant([" airplane"] * 2),
+            ),
+            "decoder_text": (
+                tf.constant([" kohli is the best"] * 2),
+                tf.constant([" kohli"] * 2),
+            ),
+        }
+
+        with self.assertRaises(ValueError):
+            self.preprocessor(input_data)
+
+    @parameterized.named_parameters(
+        ("tf_format", "tf", "model"),
+        ("keras_format", "keras_v3", "model.keras"),
+    )
+    def test_saved_model(self, save_format, filename):
+        input_data = {
+            "encoder_text": tf.constant(" airplane at airport"),
+            "decoder_text": tf.constant(" kohli is the best"),
+        }
+
+        inputs = {
+            "encoder_text": keras.Input(dtype="string", shape=()),
+            "decoder_text": keras.Input(dtype="string", shape=()),
+        }
+        outputs = self.preprocessor(inputs)
+        model = keras.Model(inputs=inputs, outputs=outputs)
+
+        path = os.path.join(self.get_temp_dir(), filename)
+        model.save(path, save_format=save_format)
+
+        restored_model = keras.models.load_model(path)
+
+        model_output = model(input_data)
+        restored_model_output = restored_model(input_data)
+
+        self.assertAllEqual(
```

Comment:
I think `assertAllClose` will handle a nested structure here, so you could just `assertAllClose(outputs, restored_outputs)`